src/components/avatar-audio-source.js

```diff
@@ -88,9 +95,34 @@ AFRAME.registerComponent("avatar-audio-source", {
 
   init() {
     this.el.sceneEl.systems["hubs-systems"].audioSettingsSystem.registerAvatarAudioSource(this);
+    // We subscribe to audio stream notifications for this peer to update the audio source
+    // This could happen in case there is an ICE failure that requires a transport recreation.
+    NAF.connection.adapter.on("stream_updated", this._onStreamUpdated, this);
     this.createAudio();
   },
 
+  async _onStreamUpdated(peerId, kind) {
+    const audio = this.el.getObject3D(this.attrName);
+    if (!audio) return;
+    const stream = audio.source.mediaStream;
+    if (!stream) return;
+
+    getOwnerId(this.el).then(async ownerId => {
+      if (ownerId === peerId && kind === "audio") {
+        // The audio stream for this peer has been updated
+        const newStream = await NAF.connection.adapter.getMediaStream(peerId, "audio").catch(e => {
+          console.error(INFO_INIT_FAILED, `Error getting media stream for ${peerId}`, e);
+        });
+
+        if (newStream) {
```

Comment:
Nice this feels much simpler than before. Important that we keep this as simple as possible since I could easily see this being a point of failure for "audio issues"